src/Core/Addon/Theme/ThemeManager.php

```diff
@@ -544,17 +555,26 @@ private function importTranslationToDatabase(Theme $theme)
      * Get all default domain from catalog.
      *
      * @param string $locale
-     * @param \PrestaShopBundle\Translation\Provider\ThemeProvider $themeProvider
+     * @param string $themeName
      *
      * @return array
      */
-    private function getDefaultDomains($locale, $themeProvider)
+    private function getDefaultDomains($locale, $themeName)
     {
+        if (null === $kernel = $this->getKernel()) {
+            return;
+        }
+
+        $providerFactory = $kernel->getContainer()->get('prestashop.translation.provider_factory');
+        /** @var \PrestaShopBundle\Translation\Provider\ThemeProvider $themeProvider */
+        $themeProvider = $providerFactory->build(
+            new ThemesType($themeName)
+        );
```

Comment:
```suggestion
        $themeProvider = $providerFactory->build(
            TranslationController::TYPE_THEMES,
            ['name' => $themeName]
        );
```

I mean something like that

Reply:
Variable unstructured arguments like `['name' => $themeName]` are to be avoided, especially when they aren't optional ;)

Reply:
I dislike using classes in factories' parameters when we have access to a const :(